engines/buried/environ/agent3_lair.cpp

```diff
@@ -0,0 +1,1007 @@
+	// Force enable frame cycling
+	((SceneViewWindow *)viewWindow)->forceEnableCycling(true);
+
+	// Disable frame caching
+	((SceneViewWindow *)viewWindow)->enableCycleFrameCache(false);
+
+	// Force open the video
+	((SceneViewWindow *)viewWindow)->changeCycleFrameMovie(_vm->getFilePath(_staticData.location.timeZone, _staticData.location.environment, SF_CYCLES));
+
+	if (((SceneViewWindow *)viewWindow)->getGlobalFlags().alRestoreSkipAgent3Initial == 1) {
+		// Start new secondary ambient
+		_vm->_sound->setSecondaryAmbientSound(_vm->getFilePath(_staticData.location.timeZone, _staticData.location.environment, 14), 64);
+
+		_staticData.cycleStartFrame = 0;
+		_staticData.cycleFrameCount = 54;
+		_frameCycleCount = _staticData.cycleStartFrame;
+
+		_passwordIndex = 0;
+		_stepDelay = 0;
+		_timerStart = g_system->getMillis();
+
+		// Empty the input queue
+		_vm->removeMouseMessages(viewWindow);
+		_vm->removeKeyboardMessages(viewWindow);
+
+		return SC_TRUE;
+	}
+
+	// Empty the input queue
+	_vm->removeMouseMessages(viewWindow);
+	_vm->removeKeyboardMessages(viewWindow);
+
+	// Make sure we have the proper cycle going on
+	_staticData.cycleStartFrame = 54;
+	_staticData.cycleFrameCount = 90;
+	_frameCycleCount = _staticData.cycleStartFrame;
+
+	// Reset the timer and cursor
+	uint32 lastTimerValue = 0;
+	Cursor oldCursor = _vm->_gfx->setCursor(kCursorWait);
+
+	// Remove some inventory items
+	((GameUIWindow *)viewWindow->getParent())->_bioChipRightWindow->changeCurrentBioChip(kItemBioChipInterface);
+	((GameUIWindow *)viewWindow->getParent())->_inventoryWindow->removeItem(kItemBioChipJump);
+	((GameUIWindow *)viewWindow->getParent())->_inventoryWindow->removeItem(kItemEnvironCart);
+
+	_vm->_gfx->setCursor(oldCursor);
+	_staticData.navFrameIndex = -1;
+	((SceneViewWindow *)viewWindow)->playSynchronousAnimation(10);
+	_staticData.navFrameIndex = 75;
+	oldCursor = _vm->_gfx->setCursor(kCursorWait);
+	viewWindow->invalidateWindow(false);
+	_vm->_gfx->updateScreen();
+
+	// Play the next Arthur speech
+	_currentSoundID = _vm->_sound->playSoundEffect(_vm->getFilePath(IDS_AGENT3_VIRUS_SOUND_BASE), 128, false, true);
+	_timerStart = g_system->getMillis();
+	lastTimerValue = g_system->getMillis();
+
+	while (!_vm->shouldQuit() && _vm->_sound->isSoundEffectPlaying(_currentSoundID)) {
+		if (g_system->getMillis() - lastTimerValue >= 50) {
+			timerCallback(viewWindow);
+			lastTimerValue = g_system->getMillis();
+		}
+
+		_vm->_sound->timerCallback();
+		_vm->yield();
+	}
+
+	_vm->_sound->stopSoundEffect(_currentSoundID);
+
+	_vm->_gfx->setCursor(oldCursor);
+	((SceneViewWindow *)viewWindow)->playSynchronousAnimation(11);
+	oldCursor = _vm->_gfx->setCursor(kCursorWait);
+
+	// Add message here
+	Common::String text;
+	if (_vm->getVersion() >= MAKEVERSION(1, 0, 4, 0))
+		text = _vm->getString(IDS_AGENT3_VIRUS_NETWORK_TEXT);
+	else
+		text = "Neuroconductor uplink connected. Network on-line.";
```

Comment:
maybe add a translation marker?

Reply:
no, this is not GUI but in-game text. The maximum which could make sense is checking if  Japanese version has a different text here.